editor/src/core/workers/parser-printer/parser-printer-parsing.ts

```diff
@@ -1847,15 +1929,17 @@ export function parseOutJSXElements(
           }
         })
 
-        // empty element fix: if the element contains only fragments, remove any empty text blocks
+        // empty element fix: if the element contains only fragments or conditionals, remove any empty text blocks
         // that may have been introduced in between due to the code formatting
         if (isRight(children)) {
           const nonEmptyTextBlockChildren = children.value.filter(
             (c) => !(isJSXTextBlock(c) && c.text.trim().length === 0),
           )
           const onlyFragments =
```

Comment:
this should then become `shouldRemoveEmptyTextBlocks` or something like that, so we also catch future cases

Reply:
true, fixed in ef257d9da3d439074e222a80c7a3fdfad2180a74